src/gui/symbology/qgsgraduatedsymbolrendererwidget.cpp

```diff
@@ -715,12 +715,15 @@ void QgsGraduatedSymbolRendererWidget::updateUiFromRenderer( bool updateCount )
     spinPrecision->setValue( method->labelPrecision() );
     cbxTrimTrailingZeroes->setChecked( method->labelTrimTrailingZeroes() );
 
-    QgsProcessingContext context;
-    for ( const auto &ppww : std::as_const( mParameterWidgetWrappers ) )
+    if ( !mBlockClassificationMethodUpdates )
     {
-      const QgsProcessingParameterDefinition *def = ppww->parameterDefinition();
-      QVariant value = method->parameterValues().value( def->name(), def->defaultValueForGui() );
-      ppww->setParameterValue( value, context );
+      QgsProcessingContext context;
+      for ( const auto &ppww : std::as_const( mParameterWidgetWrappers ) )
+      {
+        const QgsProcessingParameterDefinition *def = ppww->parameterDefinition();
+        QVariant value = method->parameterValues().value( def->name(), def->defaultValueForGui() );
+        ppww->setParameterValue( value, context );
+      }
```

Comment:
@nyalldawson I've tested your branch again and debugged the code, it looks to me a no-op: the only case in which the `mBlockClassificationMethodUpdates` is != 0 is when it is called from the constructor. 

In all other occasions it is called from `QgsGraduatedSymbolRendererWidget::classificationMethodWidgetChanged` when it is increased before the call.

I've even added an else branch and placed a breakpoint right here and it is never reached.

I must be missing something.

Reply:
I believe it's because you were testing with your fix applied too -- indeed with your fix mBlockClassificationMethodUpdates will always be zero, because it's reset immediately before the deferred timer is called to perform the classification... I'll rework to avoid that

Reply:
Ok, I've updated accordingly (at a complexity cost :upside_down_face: )